internal/pkg/warmup/warmup.go

```diff
@@ -34,13 +34,15 @@ type Warmup struct {
 	HttpHeaders              []string
 	GrpcRequests             chan grpc.Request
 	RequestDelayMilliseconds int
+	ConcurrencyTargetSeconds int
 }
 
 // Run sends requests to the target using goroutines.
 func (w Warmup) Run(hasHttpRequests bool, hasGrpcRequests bool, requestsSentCounter *int) {
 	rand.Seed(time.Now().UnixNano()) // initialize seed only once to prevent deterministic/repeated calls every time we run
 
 	var wg sync.WaitGroup
+	var sleepInterval = w.ConcurrencyTargetSeconds / w.Concurrency
```

Comment:
Thanks for the change, and sorry for the trouble :s

I think this is easier to use

Reply:
No problem at all. I created the other pr as a draft first since I was not happy with the solution with the edge cases it created and wanted opinions.